bin/src/hickory-dns.rs

```diff
@@ -654,6 +688,115 @@ where
+            "disable_tls" => config.disable_tls().to_string(),
+            "disable_udp" => config.disable_udp().to_string(),
+            "allow_networks" => config.allow_networks().len().to_string(),
+            "deny_networks" => config.deny_networks().len().to_string(),
+            "zones" => config.zones().len().to_string()
+        );
+        describe_gauge!(
+            "hickory_config_info",
+            Unit::Count,
+            "hickory config metadata"
+        );
+        hickory_config_info.set(1);
+
+        let zones_total_name = "hickory_zones_total";
+        let zones_primary_file = counter!(zones_total_name, "store" => "file", "role" => "primary");
+        let zones_secondary_file =
+            counter!(zones_total_name, "store" => "file", "role" => "secondary");
+
+        describe_counter!(
+            zones_total_name,
+            Unit::Count,
+            "number of dns zones in storages"
+        );
+
+        #[cfg(feature = "resolver")]
+        let zones_forwarder = counter!(zones_total_name, "store" => "forwarder");
+
+        #[cfg(feature = "sqlite")]
+        let (zones_primary_sqlite, zones_secondary_sqlite) = {
+            let zones_primary_sqlite =
+                counter!(zones_total_name, "store" => "sqlite", "role" => "primary");
+            let zones_secondary_sqlite =
+                counter!(zones_total_name, "store" => "sqlite", "role" => "secondary");
+            (zones_primary_sqlite, zones_secondary_sqlite)
+        };
+
+        Self {
+            #[cfg(feature = "resolver")]
+            zones_forwarder,
+            #[cfg(feature = "sqlite")]
+            zones_primary_sqlite,
+            zones_primary_file,
+            #[cfg(feature = "sqlite")]
+            zones_secondary_sqlite,
+            zones_secondary_file,
+        }
+    }
+
+    fn increment_zone_metrics(&self, zone: &ZoneConfig) {
+        match &zone.zone_type_config {
+            ZoneTypeConfig::Primary(server_config) => {
+                for store in &server_config.stores {
+                    if matches!(store, ServerStoreConfig::File(_)) {
+                        self.zones_primary_file.increment(1)
+                    }
+                    #[cfg(feature = "sqlite")]
+                    if matches!(store, ServerStoreConfig::Sqlite(_)) {
+                        self.zones_primary_sqlite.increment(1)
+                    };
+                }
+            }
+            ZoneTypeConfig::Secondary(server_config) => {
+                for store in &server_config.stores {
+                    if matches!(store, ServerStoreConfig::File(_)) {
+                        self.zones_secondary_file.increment(1)
+                    }
+                    #[cfg(feature = "sqlite")]
+                    if matches!(store, ServerStoreConfig::Sqlite(_)) {
+                        self.zones_secondary_sqlite.increment(1)
+                    };
+                }
+            }
```

Comment:
I think we can deduplicate these arms?

Reply:
Extracted the duplicated part into a function.